### PR TITLE
.net sdk api release

### PR DESCRIPTION
**Pull request by GuptaVertika**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/network/Azure.ResourceManager.Network/CHANGELOG.md [hunk 1]
```diff
@@ -1,14 +1,10 @@
 # Release History
 
-## 1.8.0-beta.1 (Unreleased)
+## 1.8.0-beta.1 (2024-04-17)
```
Comment (archerzz) [inline sdk/network/Azure.ResourceManager.Network/CHANGELOG.md:3]: Release date should be a day in future.

File: sdk/network/Azure.ResourceManager.Network/src/Generated/FirewallPolicyRuleCollectionGroupDraftData.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        /// </param>
+        internal FirewallPolicyRuleCollectionGroupDraftData(ResourceIdentifier id, string name, ResourceType? resourceType, IDictionary<string, BinaryData> serializedAdditionalRawData, string size, int? priority, IList<FirewallPolicyRuleCollectionInfo> ruleCollections) : base(id, name, resourceType, serializedAdditionalRawData)
+        {
+            Size = size;
+            Priority = priority;
+            RuleCollections = ruleCollections;
+        }
+
+        /// <summary> A read-only string that represents the size of the FirewallPolicyRuleCollectionGroupProperties in MB. (ex 1.2MB). </summary>
+        public string Size { get; }
```
Comment (archerzz) [inline sdk/network/Azure.ResourceManager.Network/src/Generated/FirewallPolicyRuleCollectionGroupDraftData.cs:47]: Is this really a string type?

File: sdk/network/Azure.ResourceManager.Network/src/Generated/PacketCaptureData.cs [hunk 3]
```diff
@@ -106,6 +110,10 @@ internal PacketCaptureData(ResourceIdentifier id, string name, ResourceType reso
         public PacketCaptureStorageLocation StorageLocation { get; }
         /// <summary> A list of packet capture filters. </summary>
         public IReadOnlyList<PacketCaptureFilter> Filters { get; }
+        /// <summary> This continuous capture is a nullable boolean, which can hold 'null', 'true' or 'false' value. If we do not pass this parameter, it would be consider as 'null', default value is 'null'. </summary>
+        public bool? ContinuousCapture { get; }
```
Comment (archerzz) [inline sdk/network/Azure.ResourceManager.Network/src/Generated/PacketCaptureData.cs:114]: Consider a cleaner boolean name like `IsCaptureContinous`

File: sdk/network/Azure.ResourceManager.Network/src/Generated/VirtualNetworkPeeringData.cs [hunk 4]
```diff
@@ -127,5 +167,13 @@ public IList<string> RemoteVirtualNetworkAddressPrefixes
         public bool? DoNotVerifyRemoteGateways { get; set; }
         /// <summary> The resourceGuid property of the Virtual Network peering resource. </summary>
         public Guid? ResourceGuid { get; }
+        /// <summary> Whether complete virtual network address space is peered. </summary>
+        public bool? PeerCompleteVnets { get; set; }
```
Comment (archerzz) [inline sdk/network/Azure.ResourceManager.Network/src/Generated/VirtualNetworkPeeringData.cs:171]: Consider a better name for this boolean property. Maybe `IsCompleteVnetPeered`?

File: sdk/network/Azure.ResourceManager.Network/src/Generated/Models/ConnectionMonitorEndpoint.cs [hunk 5]
```diff
@@ -86,15 +90,31 @@ internal ConnectionMonitorEndpoint()
...
+            set
+            {
+                if (LocationDetails is null)
+                    LocationDetails = new ConnectionMonitorEndpointLocationDetails();
+                LocationDetails.Region = value;
+            }
+        }
+
+        /// <summary> Subscription ID for connection monitor endpoint. It's an optional parameter which is being used for 'AzureArcNetwork' type endpoint. </summary>
+        public string SubscriptionId { get; set; }
```
Comment (archerzz) [inline sdk/network/Azure.ResourceManager.Network/src/Generated/Models/ConnectionMonitorEndpoint.cs:118]: What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/maps/Azure.ResourceManager.Maps/src/autorest.md?plain=1#L25).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/search/Azure.Search.Documents/src/autorest.md?plain=1#L106-L115).

File: sdk/network/Azure.ResourceManager.Network/src/Generated/Models/HeaderValueMatcher.cs [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+            Negate = negate;
+            _serializedAdditionalRawData = serializedAdditionalRawData;
+        }
+
+        /// <summary> The pattern, either fixed string or regular expression, that evaluates if a header value should be selected for rewrite. </summary>
+        public string Pattern { get; set; }
+        /// <summary> Setting this parameter to truth value with force the pattern to do a case in-sensitive comparison. </summary>
+        public bool? IgnoreCase { get; set; }
+        /// <summary> Setting this value as truth will force to check the negation of the condition given by the user in the pattern field. </summary>
+        public bool? Negate { get; set; }
```
Comment (archerzz) [inline sdk/network/Azure.ResourceManager.Network/src/Generated/Models/HeaderValueMatcher.cs:71]: rename to like `IsNegated`?

File: sdk/network/Azure.ResourceManager.Network/src/Generated/Models/PacketCaptureCreateOrUpdateContent.cs [hunk 7]
```diff
@@ -103,5 +107,9 @@ internal PacketCaptureCreateOrUpdateContent()
         public PacketCaptureStorageLocation StorageLocation { get; }
         /// <summary> A list of packet capture filters. </summary>
         public IList<PacketCaptureFilter> Filters { get; }
+        /// <summary> This continuous capture is a nullable boolean, which can hold 'null', 'true' or 'false' value. If we do not pass this parameter, it would be consider as 'null', default value is 'null'. </summary>
+        public bool? ContinuousCapture { get; set; }
```
Comment (archerzz) [inline sdk/network/Azure.ResourceManager.Network/src/Generated/Models/PacketCaptureCreateOrUpdateContent.cs:111]: same

File: sdk/network/Azure.ResourceManager.Network/src/Generated/Models/VirtualApplianceIPConfiguration.cs [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+            Properties = properties;
+            _serializedAdditionalRawData = serializedAdditionalRawData;
+        }
+
+        /// <summary> Name of the IP configuration. </summary>
+        public string Name { get; set; }
+        /// <summary> Represents a single IP configuration properties. </summary>
+        internal VirtualApplianceIPConfigurationProperties Properties { get; set; }
+        /// <summary> Whether or not this is primary IP configuration of the NIC. </summary>
+        public bool? Primary
```
Comment (archerzz) [inline sdk/network/Azure.ResourceManager.Network/src/Generated/Models/VirtualApplianceIPConfiguration.cs:69]: reanme to `IsPrimary`?

**Review by archerzz — COMMENTED** [review]
generally fine, need to double check the names for some boolean properties.

File: sdk/network/Azure.ResourceManager.Network/src/Generated/Models/HeaderValueMatcher.cs (hunk 6 above)
Comment (archerzz) [reply sdk/network/Azure.ResourceManager.Network/src/Generated/Models/HeaderValueMatcher.cs:71]: there is already a boolean `Negate` property: https://github.com/Azure/azure-sdk-for-net/blob/3bd55fdc73b443322170d53816101ef9af5ea432/sdk/network/Azure.ResourceManager.Network/src/Generated/Models/ApplicationGatewayRewriteRuleCondition.cs#L75

so better to keep it as-is

File: sdk/network/Azure.ResourceManager.Network/src/Generated/Models/ConnectionMonitorEndpoint.cs (hunk 5 above)
Comment (GuptaVertika) [reply sdk/network/Azure.ResourceManager.Network/src/Generated/Models/ConnectionMonitorEndpoint.cs:118]: megha.jain Please check this comment

File: sdk/network/Azure.ResourceManager.Network/src/Generated/FirewallPolicyRuleCollectionGroupDraftData.cs (hunk 2 above)
Comment (bewatersmsft) [reply sdk/network/Azure.ResourceManager.Network/src/Generated/FirewallPolicyRuleCollectionGroupDraftData.cs:47]: Yes, it has "MB" at the end . It is actually the "file size"

File: sdk/network/Azure.ResourceManager.Network/src/Generated/Models/ConnectionMonitorEndpoint.cs (hunk 5 above)
Comment (v-sapanchal) [reply sdk/network/Azure.ResourceManager.Network/src/Generated/Models/ConnectionMonitorEndpoint.cs:118]: Hi @archerzz ,
It's kind of guid and it's an optional parameter.

File: sdk/network/Azure.ResourceManager.Network/src/Generated/Models/VirtualApplianceIPConfiguration.cs (hunk 8 above)
Comment (v-sapanchal) [reply sdk/network/Azure.ResourceManager.Network/src/Generated/Models/VirtualApplianceIPConfiguration.cs]: Hi @archerzz , it's a nullable Boolean parameter, and we need this parameter in this format.

File: sdk/network/Azure.ResourceManager.Network/src/Generated/Models/PacketCaptureCreateOrUpdateContent.cs (hunk 7 above)
Comment (v-sapanchal) [reply sdk/network/Azure.ResourceManager.Network/src/Generated/Models/PacketCaptureCreateOrUpdateContent.cs]: Hi @archerzz , it's a nullable Boolean parameter, and we need this parameter in this format.

File: sdk/network/Azure.ResourceManager.Network/src/Generated/PacketCaptureData.cs (hunk 3 above)
Comment (v-sapanchal) [reply sdk/network/Azure.ResourceManager.Network/src/Generated/PacketCaptureData.cs]: Hi @archerzz , it's a nullable boolean parameter.

File: sdk/network/Azure.ResourceManager.Network/src/Generated/Models/ConnectionMonitorEndpoint.cs (hunk 5 above)
Comment (GuptaVertika) [reply sdk/network/Azure.ResourceManager.Network/src/Generated/Models/ConnectionMonitorEndpoint.cs]: Added

**Review by archerzz — APPROVED** [review]

**Pull request merged**